checker/src/main/java/org/checkerframework/checker/index/inequality/LessThanAnnotatedTypeFactory.java

```diff
@@ -336,4 +348,58 @@ public AnnotationMirror createLessThanQualifier(String expression) {
             return list;
         }
     }
+
+    /**
+     * Returns true if some annotation in the given list is an index annotation that is only
+     * applicable to integral types, such as @LessThan.
+     *
+     * @param annoTrees a list of annotations on a variable/method declaration; null if this type is
+     *     not from such a location
+     * @param typeTree the type whose annotations to test
+     * @return true if some annotation is an index annotation that is only applicable to integral
+     *     types
+     */
+    protected boolean containsIntegralIndexAnnotation(
+            List<? extends AnnotationTree> annoTrees, Tree typeTree) {
+        List<? extends AnnotationTree> annos =
+                TreeUtils.getExplicitAnnotationTrees(annoTrees, typeTree);
+
+        for (AnnotationTree annoTree : annos) {
+            AnnotationMirror am = TreeUtils.annotationFromAnnotationTree(annoTree);
+            if (isIntegralIndexAnnotation(am)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if type has an index annotation that is only applicable to integral types, such
+     * as @LessThan.
+     *
+     * @param type a type
+     * @return true if the type has an index annotation that is only applicable to integral types
+     */
+    protected boolean hasIntegralIndexAnnotation(AnnotatedTypeMirror type) {
+        for (AnnotationMirror am : type.getAnnotations()) {
+            if (isIntegralIndexAnnotation(am)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if the given annotation is an index annotation that is only applicable to
+     * integral types, such as @LessThan.
+     *
+     * @param am an annotation
+     * @return true if the given annotation is an index annotation that is only applicable to
+     *     integral types
+     */
+    protected boolean isIntegralIndexAnnotation(AnnotationMirror am) {
```

Comment:
Calls to this method could be replaced with calls to `AnnotatedTypeFactory#isSupportedQualifier(AnnotationMirror)`.

Reply:
That would be correct for most annotations, but the error message "Annotation %s is only applicable to integral types" would be funny for an annotation like `@SameLen` which is applicable to sequences.

Reply:
`@SameLen` is not a supported qualifier of the LessThanChecker, so `AnnotatedTypeFactory#isSupportedQualifier(AnnotationMirror)` would return false.

Reply:
I agree.  I was thinking forward to centralizing the logic in one place.  (But, I realize that was not clear; sorry about that.)